cli/cmd/sync2.go

```diff
@@ -29,18 +30,39 @@ func syncConnectionV2(ctx context.Context, cqDir string, sourceClient *clients.S
 	}
 	defer destClients.Close()
 
-	tables, err := sourceClient.GetTables(ctx)
+	allTables, err := sourceClient.GetTables(ctx)
```

Comment:
I think this line is not needed because your call `GetTables` inside `getTablesForSpec`

Reply:
We do need it because we still want Migrate to get the full list of tables. `getTablesForSpec` only returns all tables if the `GetTablesForSpec` method is not available.